code/game/gamemodes/shadowling/shadow_abilities.dm

```diff
@@ -0,0 +1,542 @@
+	var/mob/living/user
+	var/mob/living/target
+	var/thrall_ability = FALSE
+
+/obj/effect/proc_holder/spell/targeted/sling/Click()
+	var/mob/living/user = usr
+	if(!istype(user))
+		return
+	var/msg
+	if(!can_cast(user))
+		msg = "<span class='warning'>You can no longer cast [name]!</span>"
+		remove_ranged_ability(msg)
+		return
+	if(active)
+		remove_ranged_ability()
+	else
+		add_ranged_ability(user, null, TRUE)
+
+	if(action)
+		action.UpdateButtonIcon()
+
+
+/obj/effect/proc_holder/spell/targeted/sling/InterceptClickOn(mob/living/caller, params, atom/t)
+	if(!isliving(t))
+		to_chat(caller, "<span class='warning'>You may only use this ability on living things!</span>")
+		revert_cast()
+		return
+	user = caller
+	target = t
+	if(!thrall_ability && !shadowling_check(user))
+		revert_cast()
+		return
+
+/obj/effect/proc_holder/spell/targeted/sling/revert_cast()
+	. = ..()
+	remove_ranged_ability()
+
+/obj/effect/proc_holder/spell/targeted/sling/start_recharge()
+	. = ..()
+	if(action)
+		action.UpdateButtonIcon()
+
+
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//////////////////////////now that the initial types are out of the way, we can get to the abilities!//////////////////////////////
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+
+////GLARE
+/obj/effect/proc_holder/spell/targeted/sling/glare //Stuns and mutes a human target for 10 seconds
+	name = "Glare"
+	desc = "Disrupts the target's motor and speech abilities."
+	panel = "Shadowling Abilities"
+	charge_max = 300
+	human_req = 1
+	clothes_req = 0
+	action_icon_state = "glare"
+
+
+/obj/effect/proc_holder/spell/targeted/sling/glare/InterceptClickOn(mob/living/caller, params, atom/t)
+	. = ..()
+	if(target.stat)
+		to_chat(user, "<span class='warning'>[target] must be conscious!</span>")
+		revert_cast()
+		return
+	if(is_shadow_or_thrall(target))
+		to_chat(user, "<span class='warning'>You cannot glare at allies!</span>")
+		revert_cast()
+		return
+	var/mob/living/carbon/human/M = target
+	user.visible_message("<span class='warning'><b>[user]'s eyes flash a purpleish-red!</b></span>")
+	var/distance = get_dist(target, user)
+	if (distance <= 1) //Melee
+		target.visible_message("<span class='danger'>[target] suddendly collapses...</span>")
+		to_chat(target, "<span class='userdanger'>A purple light flashes across your vision, and you lose control of your movements!</span>")
+		target.Stun(100)
+		M.silent += 10
+	else //Distant glare
+		var/loss = 100 - ((distance - 1) * 18)
+		target.adjustStaminaLoss(loss)
+		target.stuttering = loss
+		to_chat(target, "<span class='userdanger'>A purple light flashes across your vision, and exhaustion floods your body...</span>")
+		target.visible_message("<span class='danger'>[target] looks very tired...</span>")
```

Comment:
This is a lot better than what Glare used to be. That said, stamina loss is inelegant because of how weird it is (pseudo-health) and the power curve over distance here is really weird, with mostly insignificant losses until about 4 tiles away, at which point the ability doesn't become worth using at all due to how ineffectual it is. `loss` also isn't properly clamped, meaning that this could cause targets to *heal* stamina damage.